fix(ci): Windows builds fail — secure-checkout path mangled by Git Bash

## Summary

All Windows binary builds fail at the checkout step because the `lgtm-hq/lgtm-ci/.github/actions/secure-checkout` composite action references a bash script via a relative path that gets mangled by Git Bash on Windows.

## Problem

The `secure-checkout` action runs a post-checkout script using a path like:

```
D:\a\_actions\lgtm-hq\lgtm-ci\v0\.github/actions/secure-checkout/../../../scripts/ci/actions/secure-checkout.sh
```

Git Bash interprets the backslashes as escape characters, producing:

```
D:a_actionslgtm-hqlgtm-civ0.github/actions/secure-checkout/../../../scripts/ci/actions/secure-checkout.sh
```

This results in exit code 127 (`No such file or directory`).

**Full error from CI logs:**

```
D:\a\_temp\4a89a2e1.sh: line 1: D:a_actionslgtm-hqlgtm-civ0.github/actions/secure-checkout/../../../scripts/ci/actions/secure-checkout.sh: No such file or directory
##[error]Process completed with exit code 127.
```

**Affected workflow chain:**
`publish-release-on-tag.yml` → `build-binary.yml` → `Build (x86_64-pc-windows-msvc)`

**Impact:** No Windows artifacts are produced for any release. The workflow shows as failed (red X on the Windows matrix entry) even though the other 4 platforms succeed.

## Proposed Solution

The root cause is in `lgtm-hq/lgtm-ci`, not this repo. Two options:

### Option A: Fix upstream in lgtm-ci (preferred)
The `secure-checkout` composite action needs Windows-compatible path handling — either quoting the script path, converting backslashes to forward slashes, or adding a platform conditional.

### Option B: Workaround in build-binary.yml
Use `actions/checkout` directly for the Windows matrix entry instead of the `secure-checkout` wrapper:

```yaml
- name: Checkout
  if: runner.os != 'Windows'
  uses: lgtm-hq/lgtm-ci/.github/actions/secure-checkout@v0

- name: Checkout (Windows)
  if: runner.os == 'Windows'
  uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
```

## References

- Example failed run: https://github.com/lgtm-hq/Rustume/actions/runs/22176963998
- Failed job ID: `64128040863`

--- a/.github/workflows/build-binary.yml
+++ b/.github/workflows/build-binary.yml
@@ -48,8 +48,16 @@ jobs:
           egress-policy: audit
 
       - name: Checkout
+        if: runner.os != 'Windows'
         # yamllint disable-line rule:line-length
         uses: lgtm-hq/lgtm-ci/.github/actions/secure-checkout@v0
+
+      - name: Checkout (Windows)
+        if: runner.os == 'Windows'
+        # yamllint disable-line rule:line-length
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          persist-credentials: false
 
       - name: Setup Rust
         # yamllint disable-line rule:line-length

